static/app/utils/array/removeAtArrayIndex.tsx

```diff
@@ -1,8 +1,6 @@
 /**
  * Remove item at `index` in `array` without mutating `array`
  */
-export default function removeAtArrayIndex<T>(array: T[], index: number): T[] {
-  const newArray = [...array];
-  newArray.splice(index, 1);
-  return newArray;
+export default function removeAtArrayIndex<T>(array: Readonly<T[]>, index: number): T[] {
+  return array.toSpliced(index, 1);
 }
```

Comment:
Seems probably OK to keep this function around, though I did just consider 'should we just kill it'